### DISCUSSION TITLE
Bug: RAG errors when no documents availabl

### DISCUSSION THREAD

**Discussion by unicorn667** [open-webui/open-webui#7598] [General]
## Installation Method
Docker compose installation using official image ghcr.io/open-webui/open-webui:main

## Environment
- **Open WebUI Version:** main
- **Ollama:** Native installation on host system
- **Operating System:** AlmaLinux
- **Browser:** Firefox

**Confirmation:**
- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
Web search functionality should work with configured search backend (Bing).

## Actual Behavior:
Web search returns empty content error:
```
[ERROR: The content provided is empty. Please ensure that there is text or data present before proceeding.]
```

## Description
**Bug Summary:**
Web search functionality fails with empty content error when using Bing as search backend. The error occurs consistently across different search queries.

## Reproduction Details
**Steps to Reproduce:**
1. Configure OpenWebUI with Bing search backend in docker-compose.yml
2. Start containers using docker-compose up -d
3. Access web interface and attempt any web search
4. Observe empty content error in UI and logs

## Logs and Screenshots
**Docker Container Logs:**
```
Fetching pages: 0it [00:00, ?it/s]
INFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db: document  e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b85
ERROR [open_webui.apps.retrieval.main] The content provided is empty. Please ensure that there is text or data present before proceeding.
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/retrieval/main.py", line 1330, in process_web_search
    save_docs_to_vector_db(docs, collection_name, overwrite=True)
  File "/app/backend/open_webui/apps/retrieval/main.py", line 799, in save_docs_to_vector_db
    raise ValueError(ERROR_MESSAGES.EMPTY_CONTENT)
ValueError: The content provided is empty. Please ensure that there is text or data present before proceeding.
```

## Additional Information
- Using docker-compose with PostgreSQL database
- Database tables are properly created and accessible
- Basic chat functionality with Ollama works correctly
- Ollama ist installed nativly directly on the server 
- Only web search functionality is affected
- Tried with different search backends (DuckDuckGo, Bing) with same result
- Test OpenWebui on my workstation with the Bing credentials that works fine. I asume there is a configuration issue with my docker network or config

I have configured the websearch inside the gui and not over the compose file

![grafik](https://github.com/user-attachments/assets/b548167e-5cf8-4732-92b5-81ef0c147c7d)

my docker compose file
Yes I know WEBUI_CHECK_UPDATE is an open point....

`version: '3.8'
services:
  postgres:
    image: postgres:14.5
    container_name: webui-postgres
    restart: unless-stopped
    environment:
      POSTGRES_DB: webui
      POSTGRES_USER: webui_user
      POSTGRES_PASSWORD: password
      POSTGRES_HOST_AUTH_METHOD: trust
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U webui_user -d webui"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - proxy-net

  open-webui:
    image: ghcr.io/open-webui/open-webui:main
    container_name: open-webui
    restart: unless-stopped
    depends_on:
      postgres:
        condition: service_healthy
    ports:
      - "3000:8080"
    environment:
      - OLLAMA_API_BASE_URL=http://host.docker.internal:11434
      - DATABASE_URL=postgresql://webui_user:password@postgres:5432/webui
      - POSTGRES_HOST=postgres
      - POSTGRES_DB=webui
      - POSTGRES_USER=webui_user
      - POSTGRES_PASSWORD=password
      - CORS_ALLOW_ORIGIN=https://myurl
      - HOST=0.0.0.0
      - PORT=8080
      - MICROSOFT_CLIENT_ID=12345456
      - MICROSOFT_CLIENT_SECRET=12345456
      - MICROSOFT_CLIENT_TENANT_ID=12345456
      - OAUTH_PROVIDERS=microsoft
      - ENABLE_OAUTH_SIGNUP=true
      - BASE_URL=https://myurl
      - WEBUI_CHECK_UPDATE=false
      - WEBUI_NOTIFY_UPDATE=false
      - WEBUI_DISABLE_UPDATE_CHECK=true
      - DISABLE_LOCAL_REGISTRATION=true
    volumes:
      - /srv/docker/openwebui/data:/app/backend/data
    extra_hosts:
      - "host.docker.internal:host-gateway"
    networks:
      - proxy-net

volumes:
  postgres_data:
    name: webui-postgres-data

networks:
  proxy-net:
    external: True`

**Comment by unicorn667** [comment]
no one with a idear or hint for this issue?


I have setup a test vm , all empty the same docker-compose file. it looks the vector db can not use the postgresql 
Here the same error but in other context i not use any external vector db...
https://github.com/open-webui/open-webui/discussions/7437



` Market News, Stock market next week: Outlook for Dec. 2-6, 2024 63a906e5fa265a72dace9e9b856a142e6d6ea330aa4bb99c2c2a76e6e1a3bb9
open-webui      | INFO  [open_webui.apps.retrieval.main] adding to collection 63a906e5fa265a72dace9e9b856a142e6d6ea330aa4bb99c2c2a76e6e1a3bb9
open-webui      | ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'encode'
open-webui      | Traceback (most recent call last):
open-webui      |   File "/app/backend/open_webui/apps/retrieval/main.py", line 854, in save_docs_to_vector_db
open-webui      |     embeddings = embedding_function(
open-webui      |                  ^^^^^^^^^^^^^^^^^^^
open-webui      |   File "/app/backend/open_webui/apps/retrieval/utils.py", line 248, in <lambda>
open-webui      |     return lambda query: embedding_function.encode(query).tolist()
open-webui      |                          ^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui      | AttributeError: 'NoneType' object has no attribute 'encode'
open-webui      | ERROR [open_webui.apps.retrieval.main] 'NoneType' object has no attribute 'encode'
open-webui      | Traceback (most recent call last):
open-webui      |   File "/app/backend/open_webui/apps/retrieval/main.py", line 1330, in process_web_search
open-webui      |     save_docs_to_vector_db(docs, collection_name, overwrite=True)
open-webui      |   File "/app/backend/open_webui/apps/retrieval/main.py", line 876, in save_docs_to_vector_db
open-webui      |     raise e
open-webui      |   File "/app/backend/open_webui/apps/retrieval/main.py", line 854, in save_docs_to_vector_db
open-webui      |     embeddings = embedding_function(
open-webui      |                  ^^^^^^^^^^^^^^^^^^^
open-webui      |   File "/app/backend/open_webui/apps/retrieval/utils.py", line 248, in <lambda>
open-webui      |     return lambda query: embedding_function.encode(query).tolist()
open-webui      |                          ^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui      | AttributeError: 'NoneType' object has no attribute 'encode'
open-webui      | INFO:     172.26.2.21:51292 - "POST /retrieval/api/v1/process/web/search HTTP/1.1" 400 Bad Request
`

**Reply by wedobetter** [reply]
Can you rename this discussion to something like: "Bug: RAG errors when no documents available"

**Comment by wedobetter** [comment]
It looks like it's trying to do doc retrieval when there's no document available. In my case I have set up the Documents embedding model but never uploaded any document in it. In fact, if I unset the embedding model, the issues goes away.

**Comment by unicorn667** [comment]
I found my issue.

that was the right hint for me
https://cuterwrite.top/en/p/integrate-open-webui-ollama-qwen25-local-rag/

Configure these correct and all works fine. 

![grafik](https://github.com/user-attachments/assets/46adc08a-ad8b-411c-b559-72f287e0ad74)
